Topic1/key1/EcommerceWebPage.java

```diff
@@ -0,0 +1,217 @@
+package Topic1.key1;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import Topic1.key3.Counter;
+import Topic1.key5.Offer;
+
+/**
+ * @author Jordan
+ * This page has the benefits below:
+ * 10% by Credit Card. 
+ * The cheapest item is for free when the user pay by Paypal.
+ * 90% of the most expensive item is free if the user pays by Cash.
+ */
+public class EcommerceWebPage {
```

Comment:
This object looks like the entry point for our services. Let's call it EcommerceService since we will not implement any presentation aspect in last week.